src/filter/index.js

```diff
@@ -10,28 +10,80 @@ turndownService.use(turndownPluginGithubFlavouredMarkdown.gfm);
 
 export const LINKS_TO_CONVERT_SELECTOR = 'a[href]:not([href^="#"])';
 
-export default async function filter(content, selector, location, filterNames, filterFunctions) {
+export default async function filter(content, { fetch: location, select: extractionSelectors = [], remove: deletionSelectors = [], filter: serviceSpecificFilters = [] }, filterFunctions) {
   const { document: webPageDOM } = new JSDOM(content).window;
 
-  if (filterNames) {
-    filterNames.forEach(filterName => {
-      // Filters work in place
-      filterFunctions[filterName](webPageDOM);
-    });
-  }
+  serviceSpecificFilters.forEach(filterName => {
+    filterFunctions[filterName](webPageDOM); // filters work in place
+  });
 
   convertRelativeURLsToAbsolute(webPageDOM, location);
```

Comment:
Apply this only at the latest stage, both for 🌱 and to ensure selectors that would rely on the `href` attribute will work as expected.